packages/nextjs/src/middlewareClient.ts

```diff
@@ -102,3 +114,228 @@ export function createMiddlewareClient<
 		}
 	});
 }
+
+class RequestCookiesAuthStorageAdapter extends CookieAuthStorageAdapter {
+	items: { [name: string]: string } = {};
+	deleteItems: { [name: string]: boolean } = {};
```

Comment:
don't really understand why it is important to maintain this extra state here - why can't we just write directly to the cookie everytime `setItem` is called here rather than requiring the developer to call the response fn returned which calls `commitCookies`?

Reply:
Potentially folks could call getUser twice, which means that the refresh would happen twice. I thought it best to just keep it in memory once set.

About the deletes, deletes are a special "command" on Set-Cookie (set empty with immediate expire) which probably needs to be specifically encoded.

Reply:
but is there any downside to just writing to the cookie directly? the cookie is also stored in memory so i'm not sure what the actual benefit is here?